### ISSUE TITLE
[REVIEW]: GIMS: Graphical Interface for Materials Simulations

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2767]
**Submitting author:** @kokookster (<a href="http://orcid.org/0000-0003-1066-6909">Sebastian Kokott</a>)
**Repository:** <a href="https://gitlab.com/gims-developers/gims" target ="_blank">https://gitlab.com/gims-developers/gims</a>
**Version:** v1.0.2
**Editor:** @jgostick
**Reviewers:** @marshallmcdonnell, @jgostick
**Archive:** <a href="https://doi.org/10.5281/zenodo.4386436" target="_blank">10.5281/zenodo.4386436</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/0fdcabe0417ccb670ce40f788b606251/status.svg)](https://joss.theoj.org/papers/0fdcabe0417ccb670ce40f788b606251)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/0fdcabe0417ccb670ce40f788b606251"><img src="https://joss.theoj.org/papers/0fdcabe0417ccb670ce40f788b606251/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/0fdcabe0417ccb670ce40f788b606251/status.svg)](https://joss.theoj.org/papers/0fdcabe0417ccb670ce40f788b606251)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://gitlab.com/gims-developers/gims" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@marshallmcdonnell & @arose, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @jgostick know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://gitlab.com/gims-developers/gims">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kokookster) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://gitlab.com/gims-developers/gims">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kokookster) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @marshallmcdonnell, @arose, @marshallmcdonnell it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.1016/j.cpc.2009.06.022 may be a valid DOI for title: Ab initio molecular simulations with numeric atom-centered orbitals
- 10.1088/0953-8984/26/36/363202 may be a valid DOI for title: Exciting: a full-potential all-electron package implementing density-functional theory and many-body perturbation theory
- 10.1016/j.commatsci.2010.05.010 may be a valid DOI for title: High-throughput electronic band structure calculations: Challenges and tools
- 10.1007/978-1-4842-2677-3_5 may be a valid DOI for title: pytest

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: :point_left:

**Comment by marshallmcdonnell** [comment]
@jgostick Small question, it looks like I've been added twice to the review above:

**Comment by jgostick** [comment]
I'm still leaning how to interact with the whedon thing.  I thought I'd finally got it right this time :-(  Anyway, it doesn't matter much, at the end of the day I just look at the check boxes above.  I did remove the second set of check boxes for you though.

**Comment by marshallmcdonnell** [comment]
Awesome, thanks!

**Comment by marshallmcdonnell** [comment]
@Kokookster really like your software package, see it as one of many needs to bring web-based solutions in the materials modeling and simulation domain! Great documentation and clean website design, very intuitive. I have went ahead and opened the following issues in the GIMS repository for my request of minor revisions. Besides that, I don't see any issues publishing in JOSS.

**Minor revisions**
https://gitlab.com/gims-developers/gims/-/issues/55
https://gitlab.com/gims-developers/gims/-/issues/56
https://gitlab.com/gims-developers/gims/-/issues/57
https://gitlab.com/gims-developers/gims/-/issues/59
https://gitlab.com/gims-developers/gims/-/issues/60
https://gitlab.com/gims-developers/gims/-/issues/61

I also added a "suggested revision" and merge requests that do not bar acceptance for me but think would be good to address / include.

**Suggested revisions*
https://gitlab.com/gims-developers/gims/-/issues/58
https://gitlab.com/gims-developers/gims/-/merge_requests/22
https://gitlab.com/gims-developers/gims/-/merge_requests/23

I figured out the installation via ["dockerizing"](https://gitlab.com/gims-developers/gims/-/merge_requests/22) and the interface worked with everything I "threw at it" :grin: 

Tests ran as expected for me (followed the GitLab CI yaml for instructions).
I encourage expanding the test suite but doing great so far. 

My biggest issues are Python application API documentation, contribution guidelines, comparison to other similar softwares in the paper, and resolving the license clash in the OrthographicTrackballControls module.

Let me know if you have questions about these issues and happy to engage in a discussion.

**Comment by jgostick** [comment]
Hi @marshallmcdonnell, thanks for a terrific review!

**Comment by Kokookster** [comment]
Hi @marshallmcdonnell,
Thanks for your kind words and your constructive review! All points are really helpful. Thanks even for fixing small typos and adding the Docker!
Will start to address your MRs, today.

**Comment by jgostick** [comment]
Hi @Kokookster it's been about 3 weeks since you checked in here...how are the revisions coming along?

**Comment by Kokookster** [comment]
Hi @jgostick,
Hopefully, I addressed all the issues opened by @marshallmcdonnell with the following MRs:
https://gitlab.com/gims-developers/gims/-/merge_requests/30
https://gitlab.com/gims-developers/gims/-/merge_requests/28
https://gitlab.com/gims-developers/gims/-/merge_requests/26
https://gitlab.com/gims-developers/gims/-/merge_requests/25
https://gitlab.com/gims-developers/gims/-/merge_requests/24
One MR is still open and waits for the go by the co-authors, but expect this by the end of this week.
Haven't got a review by @arose. Hope I didn't miss anything.

**Comment by jgostick** [comment]
Hi @arose, how is your review coming along?  The authors have been busy addressing the comments of the other reviewer so will be ready to receive yours soon.

**Comment by marshallmcdonnell** [comment]
Hello @jgostick and @Kokookster,
Sorry for the delay in response, I have been out for the past ~2 weeks and just seeing the conversation.
Yes, @Kokookster has been doing an excellent job addressing all issues I raised!
I have tried my best to monitor the merge requests and updating my checklist above.
I only have one more issue and the draft MR @Kokookster has already appears to address it.

So, yes, I will be ready for full approval once that is merged :+1: 

Thanks for your patience with me on the review (my first JOSS review) and all the hard work!

**Comment by jgostick** [comment]
It seems that @arose has disappeared. I have emailed him but no reply.  I am sorry for the delay.  I will see about another reviewer.  I am unable to review this myself.

**Comment by Kokookster** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: :point_left:

**Comment by marshallmcdonnell** [comment]
Great work, @Kokookster, thank you for addressing all my comments!
@jgostick I am satisfied that my review has been fully addressed and believe the work is ready to be published in its current state.

I understand that we are still trying to find a second reviewer and there will be edits based on the second review.
I'll be happy to have another look after any edits from the outcome.

Let me know if there is anything else needed on my end in the meantime! :+1:

**Comment by jgostick** [comment]
@whedon remove @arose as reviewer

**Comment by whedon** [comment]
OK, @arose is no longer a reviewer

**Comment by jgostick** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by jgostick** [comment]
@whedon assign @marshallmcdonnell as reviewer

**Comment by whedon** [comment]
OK, @marshallmcdonnell is now a reviewer

**Comment by jgostick** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: :point_left:

**Comment by jgostick** [comment]
Hi @Kokookster, I decided to perform the review myself since you've been waiting so long, and @marshallmcdonnell did such a thorough job. I can confirm that I am able to check off all the boxes above.  I even installed node.js and npm so that I could build your code and all went fine.  My only comment is that the installation instructions seem to wander off near the end.  The beginning mentions that two steps are needed (install npm then pip), but I was able to open the index.html file and get things basically working without the pip part.  I *Think* the python install was only required for developers so they can see any changes they make in real time without having to build the npm package everytime...is that correct?  In any case, you might want to update the readme with a bit more details in this regard.  I might also suggest that the various modules offer some way to preload some existing files, so users can poke around immediately (like you do on your demo site), but this is not necessary of course.  

So, I'm now happy with this submission and will begin the process of accepting it.

**Comment by jgostick** [comment]
@whedon check references

**Comment by jgostick** [comment]
Here is a list of what to do next:

- [ ] Double check authors and affiliations (including ORCIDs)
- [ ] Make a release of the software with the latest changes from the review and post the version number here. This is the version that will be used in the JOSS paper.
- [ ] Archive the release on Zenodo/figshare/etc and post the DOI here.
- [ ] Make sure that the title and author list (including ORCIDs) in the archive match those in the JOSS paper.

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.1016/j.cpc.2009.06.022 may be a valid DOI for title: Ab initio molecular simulations with numeric atom-centered orbitals
- 10.1088/0953-8984/26/36/363202 may be a valid DOI for title: Exciting: a full-potential all-electron package implementing density-functional theory and many-body perturbation theory
- 10.1016/j.commatsci.2010.05.010 may be a valid DOI for title: High-throughput electronic band structure calculations: Challenges and tools
- 10.1007/978-1-4842-2677-3_5 may be a valid DOI for title: pytest

INVALID DOIs

- None
```

**Comment by jgostick** [comment]
@Kokookster you'll also have to clean up the references...they are all missing doi's.  Whedon has magically suggested 4 of them, but you have 6 papers in the bibliography.

**Comment by Kokookster** [comment]
> Hi @Kokookster, I decided to perform the review myself since you've been waiting so long, and @marshallmcdonnell did such a thorough job. I can confirm that I am able to check off all the boxes above. I even installed node.js and npm so that I could build your code and all went fine. My only comment is that the installation instructions seem to wander off near the end. The beginning mentions that two steps are needed (install npm then pip), but I was able to open the index.html file and get things basically working without the pip part. I _Think_ the python install was only required for developers so they can see any changes they make in real time without having to build the npm package everytime...is that correct? In any case, you might want to update the readme with a bit more details in this regard. I might also suggest that the various modules offer some way to preload some existing files, so users can poke around immediately (like you do on your demo site), but this is not necessary of course.
> 
> So, I'm now happy with this submission and will begin the process of accepting it.

Thanks @jgostick !

The python installation is mandatory for a working GIMS application. The static part of GIMS will still work without any running flask, however the server requests will remain unanswered, which is why you don't see the preloaded geometry from the demo site. The GIMS application should look and work like the demo page, if everything is installed properly. The easiest way to install everythin is using the docker, which was added by @marshallmcdonnell (Thanks again!).

**Comment by Kokookster** [comment]
@whedon check references

**Comment by Kokookster** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cpc.2009.06.022 is OK
- 10.1088/0953-8984/26/36/363202 is OK
- 10.1088/1361-648X/aa680e is OK
- 10.1016/j.commatsci.2010.05.010 is OK
- 10.1186/1758-2946-4-17 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: :point_left:

**Comment by jgostick** [comment]
Hi @Kokookster , can you address the rest of the checkboxes in my comment above?  Get a doi, tell me the version number, double check author affiliations, etc?

**Comment by Kokookster** [comment]
Hi @jgostick,

Sorry for the delay! Here are the requested things:
* Version number: 1.0.2 (c.f. https://gitlab.com/gims-developers/gims/-/tags/1.0.2)
* DOI of the Repo-Archive: 10.5281/zenodo.4386436

I checked all the ORCIDs and Names of the authors. They should correspond to an existing person and be consistent.
Let me know if anything is missing!

**Comment by volkerblum** [comment]
Hi @jgostick and @Kokookster - Happy 2021! I stayed silent earlier since @Kokookster handled everything expertly ... just wanted to carefully ask if anything is missing for the paper from our (author) side or if we are in principle all set. No problem if all set (I thought we are), just wanted to make sure. Thank you very much for all of this! Best wishes, Volker

**Comment by jgostick** [comment]
Sorry for the delay...just emerging from vacay mode.  I think things are ready now.  I will process shortly.

**Comment by jgostick** [comment]
@whedon set 10.5281/zenodo.4386436 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4386436" target="_blank">10.5281/zenodo.4386436</a> is the archive.

**Comment by jgostick** [comment]
@whedon set v1.0.2 as version

**Comment by whedon** [comment]
OK. v1.0.2 is the version.

**Comment by jgostick** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2767 with the following error: 

 Can't find any papers to compile :-(

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cpc.2009.06.022 is OK
- 10.1088/0953-8984/26/36/363202 is OK
- 10.1088/1361-648X/aa680e is OK
- 10.1016/j.commatsci.2010.05.010 is OK
- 10.1186/1758-2946-4-17 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by jgostick** [comment]
I don't understand the failed paper compilation thing.  An EiC will drop by to finalize the acceptance, so hopefully they'll have some insight.

**Comment by jgostick** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02767/joss.02767/10.21105.joss.02767.pdf) :page_facing_up: :point_left:

**Comment by volkerblum** [comment]
Great - yes, everything sounds good! Thank you so much! Hopefully the remaining issue can be resolved. We're very excited and happy about the paper and process.

Best wishes again!
Volker

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cpc.2009.06.022 is OK
- 10.1088/0953-8984/26/36/363202 is OK
- 10.1088/1361-648X/aa680e is OK
- 10.1016/j.commatsci.2010.05.010 is OK
- 10.1186/1758-2946-4-17 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2018

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2018, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cpc.2009.06.022 is OK
- 10.1088/0953-8984/26/36/363202 is OK
- 10.1088/1361-648X/aa680e is OK
- 10.1016/j.commatsci.2010.05.010 is OK
- 10.1186/1758-2946-4-17 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2019

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2019, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1347303651300290566) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2020
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02767](https://doi.org/10.21105/joss.02767)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@marshallmcdonnell, @jgostick - many thanks for your reviews here and to @jgostick for editing this submission too! JOSS relies upon the volunteer efforts of folks like yourselves and we simply wouldn't be able to do this without you ✨.

@Kokookster - your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02767/status.svg)](https://doi.org/10.21105/joss.02767)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02767">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02767/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02767/status.svg
   :target: https://doi.org/10.21105/joss.02767
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02767/status.svg)](https://doi.org/10.21105/joss.02767)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by Kokookster** [comment]
@jgostick, @marshallmcdonnell,  @arfon Thanks for this terrific review! It was very helpful for me and the whole project! Not only this review, but all the requirements for submission are an excellent guide for bringing the software project into good shape!
